src/arrays/arraymath.jl

```diff
@@ -121,6 +121,11 @@ function \{B<:OrthonormalBasis}(op::AbstractQuMatrix{B}, vec::AbstractQuVector{B
     return QAT(div, bases(op,1))
 end
 
+# dot product of QuArray's
+dot{B<:OrthonormalBasis}(vec1::AbstractQuVector{B}, vec2::AbstractQuVector{B}) = dot(coeffs(vec1), coeffs(vec2))
+# Reference : http://en.wikipedia.org/wiki/Dot_product#Dyadics_and_matrices
+dot{B<:OrthonormalBasis}(qm1::AbstractQuMatrix{B}, qm2::AbstractQuMatrix{B}) = trace(coeffs(qm1')*coeffs(qm2))
```

Comment:
I think it is better to use `trace(coeffs(qm1'*qm2))` here. If you want you can also steal `trace` for `QuMatrix` from #32. We have to check the issue I linked to if there is a more efficient way to do this dot product.